crates/weaver_semconv/src/group.rs

```diff
@@ -114,11 +118,30 @@ impl GroupSpec {
         }
 
         // Field name is required if prefix is empty and if type is event.
-        if self.r#type == GroupType::Event && self.prefix.is_empty() && self.name.is_none() {
+        if self.r#type == GroupType::Event {
+            if self.body.is_some() && self.name.is_none() {
+                // Must have a name which is assigned to event.name for log based events
+                errors.push(Error::InvalidGroup {
+                    path_or_url: path_or_url.to_owned(),
+                    group_id: self.id.clone(),
+                    error: "This group contains an event type with a body definition but the name is not set.".to_owned(),
+                });
+            }
+            if self.name.is_none() && self.prefix.is_empty() {
```

Comment:
cc @lmolkova - For concerns on prefix.

Reply:
FYI - This is really here to "handle" all of the existing semconv events, as there are several existing `events` that don't have a name and instead use the `prefix` as the name.

Reply:
they are gone now and nothing is allowed to add a prefix as of last version https://github.com/open-telemetry/semantic-conventions/pull/1293 :)

But it's a limitation on the semconv side and tooling is not aware of it (yet). We might remove prefix from the schema as well - https://github.com/open-telemetry/weaver/issues/263.

There is also a chance that weaver is used with older versions of semconv.

TL:DR: we should probably to support prefix for the time being, but it's unlikely it'll ever be used on events.

Reply:
> we should probably support prefix for the time being, but it's unlikely it'll ever be used on new events.

Agree, as if I remove this then this would also force that the released version of weaver (with this change) will only work with the versions of semconv that has them removed.